### PR TITLE
Configurability

### PR DESCRIPTION
**Pull request by Carreau**
Sync branch just for me. 

I'll turn that into a proper PR later.

### CODE REVIEW

**Comment by Carreau** [comment]
Need ipython/ipython#7755 and ping @rgbkrk.

**Comment by KesterTong** [comment]
Nice job getting this to work.  It's annoying to have to do that dance with promises so we can wait until the Contents object is loaded before starting authorization, but at least it works.

File: jupyterdrive/gdrive/gapi_utils.js [hunk 1]
```diff
@@ -208,21 +224,37 @@ define(function(require) {
...
+      _handle.resolve(conf);
+    }
+
     /**
      * Promise fullfilled when gapi is loaded, and authorization is complete.
      */
-    var gapi_ready = load_gapi_1().then(load_gapi_2).then(authorize);
+    var load_gapi = load_gapi_1().then(load_gapi_2)
+    var gapi_ready = Promise.all([load_gapi, _conf_prm]).then(
+      function(resolve,reject){
```
Comment (KesterTong) [inline jupyterdrive/gdrive/gapi_utils.js]: This should be ```function(args)``` or similar.  ```resolve``` and ```reject``` are the args the the anonymous function in the promise function, and are callbacks used to resolve/reject the promise.  But ```then``` takes just one argument, which is whatever the Promise was resolved with.  The function is correct though, it's just the names and extra argument are confusing.

Comment (Carreau) [inline jupyterdrive/gdrive/gapi_utils.js]: Yes, make sens. 

Probably due to some leftover experimentations.
-- 
M

**Comment by KesterTong** [comment]
Is this working yet?  If so I'll test it out and do a final review.

**Comment by Carreau** [comment]
Yes sure it should. Despite the PR title that I'll update.

**Comment by KesterTong** [comment]
I can't test this on my desktop because I can't install it without the fixes in mixedcontents branch.  Do you think it's time to merge mixedcontents with master?  If so, can you do that then rebase this onto master?

**Comment by Carreau** [comment]
Need rebase.

**Comment by Carreau** [comment]
Rebased and merged.

**Comment by Carreau** [comment]
squashed, not merged.

File: jupyterdrive/gdrive/gapi_utils.js [hunk 2]
```diff
@@ -208,21 +223,38 @@ define(function(require) {
...
         }
     };
 
+    var _handle = {};
+    var _conf_prm = new Promise(function(resolve){
+        _handle.resolve = resolve;
+    })
+
+
+    var config = function(conf){
```
Comment (KesterTong) [inline jupyterdrive/gdrive/gapi_utils.js]: Can you add a comment like "calling config with conf, results in the promise _conf_prm being resolved with conf.  This then triggers the rest of the gapi loading".

Comment (Carreau) [inline jupyterdrive/gdrive/gapi_utils.js]: Done.

**Comment by KesterTong** [comment]
Ok works fine, just one small comment above.  How about we just keep APP_ID with the same name, but set its initial value to the empty string, and set it from config when config() is called?

**Comment by Carreau** [comment]
> Ok works fine, just one small comment above. Also, how about we just keep APP_ID with the same name, but set its initial value to the empty string, and set it from config when config() is called?

We could do that, but it is exported, so I'm not sure how it behaves.
So I don't want to bake in a configurable that I'm not sure works correctly.

**Comment by KesterTong** [comment]
Ah good point.  We'll do it later then.

**Pull request merged**